onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -54,14 +137,34 @@ static bool IsCastTo(const Node* node, TensorProto_DataType data_type) {
   return false;
 }
 
+// Check whether the node-arg element type is same the specified data type
 static bool IsType(const NodeArg& node_arg, TensorProto_DataType data_type) {
   return node_arg.TypeAsProto()->tensor_type().elem_type() == data_type;
 }
 
-// InsertCastNodes
-// Insert a new Cast node after each NodeArg in the require_cast map, feeding the nodes in the vector mapped to
-// the NodeArg. The other consumers of the NodeArg will not be changed. The cast node is FLOAT16 if is_fp16 is True
-// and FLOAT otherwise. This funtion fixes the graph edges in addition to inserting the cast nodes.
+/* InsertCastNodes
+* Insert a new Cast node after each NodeArg in the require_cast map, feeding the nodes (consumer) in the vector mapped to
+* the NodeArg. The other consumers of the NodeArg will not be changed. The cast node is FLOAT16 if is_fp16 is True
+* and FLOAT otherwise. This function fixes the graph edges in addition to inserting the cast nodes.
+*
+* In the following example only the first two consumers, Opcode0 and Opcode1 get casted and the third consumer Opcode2 does not.
```

Comment:
do we make sure that the other consumers( that are not casted) are getting the valid type somewhere?

Reply:
The type of the other consumers remains unchanged.